apps/web/src/dialogs/command-palette/command-palette-dialog.tsx

```diff
@@ -0,0 +1,317 @@
+                                    : "icon"
+                                }
+                              />
+                            )}
+                            {["note", "notebook", "reminder", "tag"].includes(
+                              command.type
+                            ) ? (
+                              <Text
+                                className="chip"
+                                sx={{
+                                  px: 1,
+                                  borderRadius: "4px",
+                                  border: "1px solid",
+                                  borderColor: "border"
+                                }}
+                              >
+                                <Highlighter
+                                  text={command.title}
+                                  query={query}
+                                />
+                              </Text>
+                            ) : (
+                              <Highlighter text={command.title} query={query} />
+                            )}
+                          </Button>
+                          {command.group === "recent" && (
+                            <Button
+                              title="Remove from recent"
+                              onClick={(e) => {
+                                e.stopPropagation();
+                                CommandPaletteUtils.removeRecentCommand(
+                                  command.id
+                                );
+                                setCommands(
+                                  CommandPaletteUtils.defaultCommands()
+                                );
+                              }}
+                              variant="icon"
+                              sx={{
+                                p: 1,
+                                mr: 1
+                              }}
+                            >
+                              <Cross size={14} />
+                            </Button>
+                          )}
+                        </Flex>
+                      ))}
+                    </Flex>
+                  </Flex>
+                ))}
+              </Flex>
+            </ScrollContainer>
+          </>
+        </Flex>
+        <Flex
+          sx={{ flexDirection: "row", bg: "hover", justifyContent: "center" }}
+        >
+          <Text variant="subBody" sx={{ m: 0.5 }}>
+            <kbd>{">"}</kbd> for command mode · remove <kbd>{">"}</kbd> for
+            search mode · <kbd>⏎</kbd> to select · <kbd>↑</kbd>
+            <kbd>↓</kbd> to navigate
+          </Text>
+        </Flex>
+      </Dialog>
+    );
+  }
+);
+
+function Highlighter({ text, query }: { text: string; query: string }) {
+  const queryClean = query.startsWith(">")
+    ? query.slice(1).trim()
+    : query.trim();
+  const result =
+    queryClean.length > 0
+      ? db.lookup.fuzzy(queryClean, text, {
```

Comment:
This is redundant work. Ideally, there should be a function in `lookup` that takes a list of objects (following whatever interface), a query, and suffix/prefix and returns a list of results that are already wrapped in a suffix/prefix so we can skip all this extra work.

Reply:
I'm unsure if the changes are still in line with what you have in mind, do take a look please

Reply:
This will do for now, I think.